frontend/public/components/cluster-settings/openid-idp-form.tsx

```diff
@@ -0,0 +1,272 @@
+    const mappingMethods = {
+      'claim': 'Claim',
+      'lookup': 'Lookup',
+      'add': 'Add',
+    };
+    return <div className="co-m-pane__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form onSubmit={this.submit} name="form" className="co-m-pane__body-group co-m-pane__form">
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <p className="co-m-pane__explanation">
+          Integrate with an OpenID Connect identity provider using an Authorization Code Flow.
+        </p>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="name">Name</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={name}
+            aria-describedby="oid-name-help"
+            id="name"
+            required />
+          <p className="help-block" id="oid-name-help">
+            Unique name of the new identity provider. This cannot be changed later.
+          </p>
+        </div>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="tag">Mapping Method</label>
+          <Dropdown dropDownClassName="dropdown--full-width" items={mappingMethods} selectedKey={mappingMethod} title={mappingMethods[mappingMethod]} onChange={this.mappingMethodChanged} />
+          <div className="help-block" id="mapping-method-description">
+            { /* TODO: Add doc link when available in 4.0 docs. */ }
+            Specifies how new identities are mapped to users when they log in.
+          </div>
+        </div>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="clientID">ClientID</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={clientID}
+            aria-describedby="idp-clientid-help"
+            id="clientID"
+            required />
+        </div>
+        <div className="form-group">
+          <DroppableFileInput
+            onChange={this.clientSecretFileChanged}
+            inputFileData={clientSecretFileContent}
+            id="clientsecret-file"
+            label="Client secret"
+            isRequired
+            hideContents />
+        </div>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="issuer">Issuer URL</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={issuer}
+            aria-describedby="idp-issuer-help"
+            id="issuer"
+            required />
+        </div>
+        <div className="co-form-section__separator"></div>
+        <label className="control-label">Claims</label>
+        <p className="co-help-text">The first non-empty claim is used. At least one claim is required.</p>
```

Comment:
This text doesn't make sense if we only let you add one

Reply:
@spadgett based on this text, I would think these should be required fields, right?